### PR TITLE
KFSPTS-25387: Changed the following lookups and inquiries to use the new frameworks: ContractGrantReportingCode, AppropriationAccount, AccountReversion, AccountGlobalSearch.

### PR DESCRIPTION
**Pull request by nkimble**
**This is NOT ready to be merged yet.** 
Trying to ensure unit tests also work as more difficult framework adjustments are made.

This set of changes has 
- Appropriation Account – done
- Contract Grant Reporting Code - done
- Account Global Search – not correct: need to make more adjustments due to the following documents not operating the same with these changes in place : Account Delegate Global Maintenance Document, Account Global Maintenance Document, Sub Object Code Maintenance Document
- Account Reversion - University fiscal year search is not presenting system options correctly, missing large amount of lookup/search data.

There are corresponding SQL changes that are also needed when the final review and merged is performed.

### CODE REVIEW

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountGlobalSearch.xml [hunk 1]
```diff
@@ -1,49 +1,78 @@
...
-    <property name="control">
-        <bean parent="CheckboxControlDefinition" />
-    </property>
-    <property name="formatterClass" value="org.kuali.kfs.core.web.format.BooleanFormatter"/>
-  </bean>
-  
-  <!-- Business Object Lookup Definition -->
+    <!-- Attribute Definitions -->
+    <bean id="AccountGlobalSearch-orgHierarchy" parent="AccountGlobalSearch-orgHierarchy-parentBean"/>
+    <bean id="AccountGlobalSearch-orgHierarchy-parentBean" abstract="true" parent="activeIndicatorAttributeDefinition"
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountGlobalSearch.xml:22]: I made this suggested change as well as the one indicated on line29 below but I was still not able to get the "Search by Organization Hierarchy" radio buttons "Yes/No/Both" to display on the Account Global Lookup Search dialog. 
Any additional insight would be appreciated.

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountGlobalSearch.xml [hunk 2]
```diff
@@ -1,49 +1,78 @@
...
+          p:name="useOrgHierarchy"
+          p:label="Search by Organization Hierarchy"
+          p:shortLabel="useOrgHierarchy"
+          p:required="false">
+    </bean>
+
+    <bean id="AccountGlobalSearch-orgHierarchy-defaultNo" parent="AccountGlobalSearch-orgHierarchy" p:defaultValue="N"/>
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountGlobalSearch.xml:29]: See comment above (line 22).

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountGlobalSearch.xml [hunk 3]
```diff
@@ -1,49 +1,78 @@
...
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans" 
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+    xmlns:p="http://www.springframework.org/schema/p" 
+    xsi:schemaLocation=" http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-2.0.xsd">
+
+    <import resource="classpath:org/kuali/kfs/sys/sys-lookup-beans.xml"/>
+    <import resource="classpath:edu/cornell/kfs/sys/cu-sys-attribute-beans.xml"/>
+    
+    <bean id="AccountGlobalSearch" parent="Account"
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountGlobalSearch.xml:10]: I have not made a change yet for this item. Waiting until I can get the lookup screen to present correctly first.

**Review by nkimble — COMMENTED** [review]
These are overall question regarding what I am seeing when comparing INTDEV with DEV/TEST/PROD:

Should the active indicators on the lookup screens be “Status” with radio buttons “Active”, “Inactive”, “All”  
OR
should the lookup screens have “Active Indicator” with radio buttons “Yes”, “No”, “Both”?
 
I am seeing both versions based upon different lookups.
Should we note which version they should be in the instructions page to make it clear?

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountGlobalSearch.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountGlobalSearch.xml]: It looks like `AccountGlobalSearch` is being used as a multi-value lookup on the Account Global document. To get the Account Global Lookup search to use the new framework (and potentially the search-by-org-hierarchy field), you need to configure the "newLookup" attribute on the Account Global doc's related maintenance collection bean. I've updated the related Confluence page with an example of how this is being configured in base code.

Sorry for not noticing this sooner.

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AppropriationAccount.xml [hunk 4]
```diff
@@ -1,164 +1,149 @@
...
+          p:shortLabel="Project Number"
+          p:maxLength="15"
+          p:validationPattern-ref="FifteenCharacterAlphaNumericValidation"
+          p:control-ref="FifteenCharacterTextControl">
+    </bean>
+
+    <bean id="AppropriationAccount-active" parent="AppropriationAccount-active-parentBean"/>
+    <bean id="AppropriationAccount-active-parentBean" abstract="true" parent="GenericAttributes-activeIndicator"
+          p:name="active"
+          p:control-ref="GenericAttributes-genericBooleanYNBoth-lookupControl">
```
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AppropriationAccount.xml:85]: Regarding your question about active-flag terminology, it looks like older-style lookups generally use the "Active Indicator: Yes/No/Both" setup, whereas newer-style lookups generally use the "Status: Active/Inactive/All" setup. I would suggest adjusting these lookups to follow the appropriate terminology, based on whether the beans are for the old or new lookup frameworks.

I also added a related note to the Confluence instructions.

**Review by cah292 — CHANGES_REQUESTED** [review]
Here are a few more items I noticed when reviewing the latest revisions. I also added a separate note on how to potentially fix the Account Global Search issue you were experiencing.

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversion.xml [hunk 5]
```diff
@@ -1,218 +1,293 @@
...
+                <ref bean="AccountReversion-budgetReversionChartOfAccountsCode"/>
+                <ref bean="AccountReversion-budgetReversionAccountNumber"/>
+                <ref bean="AccountReversion-cashReversionFinancialChartOfAccountsCode"/>
+                <ref bean="AccountReversion-cashReversionAccountNumber"/>
+                <ref bean="AccountReversion-carryForwardByObjectCodeIndicator"/>
+                <ref bean="activeIndicatorAttributeDefinition"/>
+            </list>
+        </property>
+        
+        <!-- LEGACY ITEMS lookupFields and resultFields - Needed to have System Options Lookup function correctly. -->
```
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversion.xml:265]: I believe it should be okay to proceed with using the newer-style System Options lookup in spite of its reduced search field count. There's only one System Options BO per fiscal year (and the fiscal year is the primary key), so users should easily be able to find and select an appropriate entry.

**Pull request closed without merging**